azurerm/internal/services/hardwaresecuritymodule/tests/resource_arm_dedicated_hardware_security_module_test.go

```diff
@@ -0,0 +1,287 @@
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testAccAzureRMDedicatedHardwareSecurityModule_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMDedicatedHardwareSecurityModule_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_dedicated_hardware_security_module" "import" {
+  name                = azurerm_dedicated_hardware_security_module.test.name
+  resource_group_name = azurerm_dedicated_hardware_security_module.test.resource_group_name
+  location            = azurerm_dedicated_hardware_security_module.test.location
+  sku_name            = azurerm_dedicated_hardware_security_module.test.sku_name
+  stamp_id            = azurerm_dedicated_hardware_security_module.test.stamp_id
+
+  network_profile {
+    network_interface_private_ip_addresses = azurerm_dedicated_hardware_security_module.test.network_profile[0].network_interface_private_ip_addresses
+    subnet_id                              = azurerm_dedicated_hardware_security_module.test.network_profile[0].subnet_id
+  }
+}
+`, config)
+}
+
+func testCheckAzureRMDedicatedHardwareSecurityModuleExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).HardwareSecurityModules.DedicatedHsmClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("dedicated hardware security module not found: %s", resourceName)
+		}
+		id, err := parse.DedicatedHardwareSecurityModuleID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Dedicated HardwareSecurityModule %q does not exist", id.Name)
+			}
+			return fmt.Errorf("bad: Get on HardwareSecurityModules.DedicatedHsmClient: %+v", err)
+		}
+		return nil
+	}
+}
+
+func testCheckAzureRMDedicatedHardwareSecurityModuleDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).HardwareSecurityModules.DedicatedHsmClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_dedicated_hardware_security_module" {
+			continue
+		}
+		id, err := parse.DedicatedHardwareSecurityModuleID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on HardwareSecurityModules.DedicatedHsmClient: %+v", err)
+			}
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccAzureRMDedicatedHardwareSecurityModule_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-hsmrg-%d"
```

Comment:
all rg should start with acctestRG
```suggestion
  name     = "acctestRG-hsm-%d"
```

Reply:
updated